test/SAMtags/parse_mm.pl

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/perl -w
+
+# Copyright (c) 2020 Genome Research Ltd.
+# Author: James Bonfield <jkb@sanger.ac.uk>
+# 
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+# 
+#    1. Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+# 
+#    2. Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+# 
+#    3. Neither the names Genome Research Ltd and Wellcome Trust Sanger
+# Institute nor the names of its contributors may be used to endorse or promote
+# products derived from this software without specific prior written permission.
+# 
+# THIS SOFTWARE IS PROVIDED BY GENOME RESEARCH LTD AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL GENOME RESEARCH LTD OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use strict;
+
+# Complement a sequence
+sub comp {
+    my ($seq)=@_;
+    $seq =~ tr/ACGTUMRWSYKVHDBN/TGCAAKYWSRMBDHVN/;
+    return $seq;
+}
+
+# Reverse complement a sequence
+sub rc {
+    my ($seq)=@_;
+    $seq = reverse($seq);
+    return comp($seq);
+}
+
+my $nseq = 0;
+while (<>) {
+    chomp($_);                                 # trim newline
+    next if /^\@/;
+
+    my ($qname,$flag,$rname,$pos,$mapq,$cigar,$rnext,
+	$pnext,$tlen,$seq,$qual,@aux) = split("\t",$_);
+    my $dir = $flag & 0x10 ? "-" : "+";
+    my ($mod_str)="@aux"=~m/MM:Z:(\S+)/;
```

Comment:
Could these two patterns be converted to allow for the probationary tags with lower case "m" and "l" in order to validate the initial versions of this tag?

Reply:
Yes. Thanks for the suggestion.